packages/foundation/src/createComponent.tsx

```diff
@@ -52,17 +52,18 @@ export function createComponent<
   ) => {
     const settings: ICustomizationProps<TViewProps, TTokens, TStyleSet> = _getCustomizations(
       options.displayName,
+      // eslint-disable-next-line react-hooks/rules-of-hooks
```

Comment:
I don't think that is another way to fix this as the plugin raises a warning because `useContext()` is used outside of a real component

Reply:
This is a real component, it's just not named as one, which confuses the rule.

Reply:
If it acts like the TSLint version, you can fix it by PascalCasing the variable `Result`